measure.rb

```diff
@@ -1888,10 +1900,19 @@ def self.add_hot_water_and_appliances(runner, model, building, weather, spaces)
           capacity_kbtuh = 100000000.0
           oncycle_power = 0.0
           offcycle_power = 0.0
+
+          if has_desuperheater
+            if not related_hvac_list.include? relatedhvac
+              related_hvac_list << relatedhvac
+              coil_object = get_desuperheatercoil(@hvac_map, relatedhvac, sys_id)
+            else
+              fail "RelatedHVACSystem '#{relatedhvac}' for water heating system '#{sys_id}' is already attached to another water heating system."
+            end
+          end
```

Comment:
Please try to avoid duplicate code two places :) Also, because we will ultimately allow desuperheater with any water heating system, it seems like the code can simply be inserted right after has_desuperheater and relatedhvac are retrieved.

Reply:
I agree that we could shift this part out of the water heater type conditions. But for E+ capability, the desuperheater is not allowed together with heat pump water heater (because they both require connecting with tank source sides which is only one space available currently) I am wondering if any check need to be added for this concern.

Reply:
So, this makes me think that we should add test files for desuperheater with:
1. Multispeed cooling coil
2. GSHP cooling coil
3. HPWH
Let's add them as *.xml.skip, so that they don't run on the CI, since they will fail. But you should be able to at least demonstrate that the first two run locally with your E+ PR, right? And we'll continue to investigate the HPWH question.

Reply:
Yes, I could definitely add the .skip and start with testing first two.